glyphs2fontir/src/source.rs

```diff
@@ -1603,28 +1604,52 @@ impl Work<Context, WorkId, Error> for ColorPaletteWork {
     }
 
     fn exec(&self, context: &Context) -> Result<(), Error> {
-        // Since we're only [for now, baby steps] producing one palette we can dedup it
-        let palette = self
+        // Directly declared palette(s)
+        let mut palettes = if let Some(declared_palettes) = self
             .font_info
             .font
-            .glyphs
-            .values()
-            .flat_map(|g| {
-                g.layers
-                    .iter()
-                    .filter(|l| l.attributes.color)
-                    .flat_map(|l| l.shapes.iter())
-                    .flat_map(|s| s.attributes().colors())
-            })
-            .map(|c| Color {
-                r: c.r as u8,
-                g: c.g as u8,
-                b: c.b as u8,
-                a: c.a as u8,
-            })
-            .collect::<Vec<_>>();
-        debug!("{} color palette entries", palette.len());
-        match ColorPalettes::new(vec![palette]) {
+            .custom_parameters
+            .color_palettes
+            .as_ref()
+        {
+            declared_palettes
+                .iter()
+                .map(|raw_palette| {
+                    raw_palette
+                        .iter()
+                        .map(|c| to_ir_color(*c))
+                        .collect::<Vec<_>>()
+                })
+                .collect::<Vec<_>>()
+        } else {
+            Vec::new()
+        };
+
+        if palettes.is_empty() {
+            palettes.push(Vec::new());
+        }
+
+        // Plus any colors declared directly on glyphs
+        palettes[0].extend(
```

Comment:
this is only modifying the first, adding to all would make them same length at least

Reply:
Done